authLdap.php

```diff
@@ -310,6 +326,48 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
 		$loggedInUser = $mapper($loggedInUser);
 	}
 
+	// Set global variable to distinguish LDAP and non-LDAP logins for authorization flow
+	global $authLDAPisLdapLogin;
+	$authLDAPisLdapLogin = true;
+
+	return $loggedInUser;
+}
+
+/**
+ * This method authorizes a user based on their LDAP group memberships
+ * after logging in, including External users if enabled.
+ * 
+ *
+ * @param null|WP_User|WP_Error
+ * @param string $username
+ * @return WP_User object
+ * @conf boolean authLDAP true, if authLDAP should be used, false if not. Defaults to false
+ * @conf boolean authLDAPExternalUsers true, if LDAP should be used for authorization of non-LDAP logins, false if not. Defaults to false
+ * @conf boolean authLDAPLocalWithExternal true, if local users are still allowed when authorizing external users, false if not. Defaults to false
+ * @conf string authLDAPFilter LDAP filter to use to find correct user, defaults to '(uid=%s)'
+ * @conf string authLDAPUidAttr LDAP attribute containing user id (the username we log on with), defaults to 'uid'
+ * @conf string authLDAPDefaultRole default role for authenticated user, defaults to ''
+ * @conf boolean authLDAPGroupEnable true, if we try to map LDAP groups to Wordpress roles
+ * @conf boolean authLDAPGroupOverUser true, if LDAP Groups have precedence over existing user roles
+ */
+function authLdap_authorization($loggedInUser, $username){
+	global $authLDAPisLdapLogin;
+	// Don't trigger if this isn't an LDAP login AND External Users aren't enabled
+	if (!($authLDAPisLdapLogin === true) && !authLdap_get_option('ExternalUsers')) {
+		authLdap_debug('User ineligible for LDAP Authorization ');
+		return $loggedInUser;
+	}
+
+	// If this isn't already an LDAP user, force a reset in the LDAP server list, or LDAP lookups will fail
+	if (!($authLDAPisLdapLogin === true)) {
+		$ldapServerList = authLdap_get_server("reset");
+		$ldapServerList->bind();
```

Comment:
I'd ratehr move the logic into the `Authorize` class to have everything deciding whether the class should do something actually within that class. That makes future maintenance easier as there is only one place to modify when conditions should change.

Reply:
As a matter of principle, I agree with you here. I was (and still am) struggling with how to make that happen without building in two-way dependency loops between `Authorize` and the core plugin script (which provides `authLdap_get_server` and builds the LDAP connections). Frankly, I'm not thrilled with the idea of using a global either, so I want to take some time to think about that piece. But I'm definitely open to suggestions on both fronts.

Reply:
Those dependencies are a completely separate topic that I will address in a separate PR later. That will then also get rid of the WordPress functions scattered throughout the code and use "propper" dependency injection etc.

So for now: Don't worry about that.